src/core/ddsi/include/dds/ddsi/q_radmin.h

```diff
@@ -98,7 +98,7 @@ struct nn_rmsg {
 
   struct nn_rmsg_chunk chunk;
 };
-#define NN_RMSG_PAYLOAD(m) ((unsigned char *) (&(m)->chunk + 1))
+#define NN_RMSG_PAYLOAD(m) ((unsigned char *) (m + 1))
```

Comment:
At the moment this should be ok because ``chunk``'s alignment should be >= ``struct nn_rmsg``'s alignment. One would expect this to always be the case, but this changes will break things in a very nasty way if one day it isn't.

So it should then at least have a static assertion added to ensure ``sizeof(m)`` equals ``offsetof(m,chunk)+sizeof(chunk)``

Reply:
I will add the static assertion.

Reply:
I will add the static assertion in the function where the macro is used.